addons/analytic/models/analytic_account.py

```diff
@@ -174,3 +174,14 @@ def convert(amount, from_currency):
                 account.debit = -data_debit.get(account.id, 0.0)
                 account.credit = data_credit.get(account.id, 0.0)
                 account.balance = account.credit - account.debit
+
+    @api.ondelete(at_uninstall=False)
+    def _unlink_except_analytic_items(self):
+        """ Check if analytic items (e.g. account moves, analytic simulations) are associated to the account """
+        domain = [('company_id', 'in', [False] + self.env.companies.ids)]
+        for plan, accounts in self.grouped('plan_id').items():
+            if self.env['account.analytic.line']._read_group(
```

Comment:
why a `_read_group`? Isn't `search_count(domain, limit=1)` enough?
We don't need to fetch the data, just check for existence.